resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -5,293 +5,1124 @@
+    <xs:complexType name="locatorListType">
+        <xs:sequence  minOccurs="0" maxOccurs="unbounded">
+            <xs:element name="locator" minOccurs="1" maxOccurs="unbounded">
+                <xs:complexType>
+                    <xs:choice>
+                        <xs:element name="udpv4" type="udpv4LocatorType" minOccurs="0" maxOccurs="unbounded"/>
+                        <xs:element name="udpv6" type="udpv6LocatorType" minOccurs="0" maxOccurs="unbounded"/>
+                        <xs:element name="tcpv4" type="tcpv4LocatorType" minOccurs="0" maxOccurs="unbounded"/>
+                        <xs:element name="tcpv6" type="tcpv6LocatorType" minOccurs="0" maxOccurs="unbounded"/>
+                    </xs:choice>
+                </xs:complexType>
+            </xs:element>
+        </xs:sequence>
+    </xs:complexType>
+
+    <!--External Locator list:
+        ├ udpv4ExternalLocator [0~*],
+        └ udpv6ExternalLocator [0~*]-->
+    <xs:complexType name="externalLocatorListType">
+        <xs:sequence minOccurs="0" maxOccurs="unbounded">
+            <xs:choice>
+                <xs:element name="udpv4" type="udpv4ExternalLocatorType" minOccurs="0" maxOccurs="unbounded"/>
+                <xs:element name="udpv6" type="udpv6ExternalLocatorType" minOccurs="0" maxOccurs="unbounded"/>
             </xs:choice>
         </xs:sequence>
     </xs:complexType>
 
-    <xs:complexType name="writerTimesType">
-        <xs:all minOccurs="0">
-            <xs:element name="initialHeartbeatDelay" type="durationType" minOccurs="0"/>
-            <xs:element name="heartbeatPeriod" type="durationType" minOccurs="0"/>
-            <xs:element name="nackResponseDelay" type="durationType" minOccurs="0"/>
-            <xs:element name="nackSupressionDuration" type="durationType" minOccurs="0"/>
+    <!--Properties policy:
+        └ properties   [1]
+            └ property [1~*] -->
+    <xs:complexType name="propertyPolicyType">
+        <xs:all>
+            <xs:element name="properties" minOccurs="1" maxOccurs="1">
+                <xs:complexType>
+                    <xs:sequence minOccurs="1" maxOccurs="unbounded">
+                        <xs:element name="property" type="propertyType" minOccurs="1" maxOccurs="unbounded"/>
+                    </xs:sequence>
+                </xs:complexType>
+            </xs:element>
         </xs:all>
     </xs:complexType>
 
-    <xs:complexType name="readerTimesType">
-        <xs:all minOccurs="0">
-            <xs:element name="initialAcknackDelay" type="durationType" minOccurs="0"/>
-            <xs:element name="heartbeatResponseDelay" type="durationType" minOccurs="0"/>
+    <!--Duration Type:
+        ├ sec     [uint32 OR "DURATION_INFINITY" OR "DURATION_INFINITE_SEC"]  REQ,
+        └ nanosec [uint32 OR "DURATION_INFINITY" OR "DURATION_INFINITE_NSEC"] REQ -->
+    <xs:complexType name="durationType">
+        <xs:all>
+            <xs:element name="sec" default="0" minOccurs="1" maxOccurs="1">
```

Comment:
I have tried erasing the value inside a "sec" tag and inside a "nanosec" tag, and the validator says it is valid.